tree/ntuple/v7/src/RNTuple.cxx

```diff
@@ -42,7 +42,19 @@ void ROOT::Experimental::RNTuple::Streamer(TBuffer &buf)
       auto offCkData = offClassBuf + sizeof(UInt_t) + sizeof(Version_t);
       auto checksum = XXH3_64bits(buf.Buffer() + offCkData, lenCkData);
 
-      buf.ReadClassBuffer(RNTuple::Class(), this, classVersion, offClassBuf, bcnt);
+      buf >> fVersionEpoch;
+      buf >> fVersionMajor;
+      buf >> fVersionMinor;
+      buf >> fVersionPatch;
+      buf >> fSeekHeader;
+      buf >> fNBytesHeader;
+      buf >> fLenHeader;
+      buf >> fSeekFooter;
+      buf >> fNBytesFooter;
+      buf >> fLenFooter;
+      if (classVersion > 4)
+         buf >> fMaxKeySize;
+      buf >> fChecksum;
```

Comment:
That's much better than it was before but still not forward compatible. After reading the last known member, we should skip all but the last 8 bytes, which we stream into the checksum member. I think we should also first make a check on the buffer size to make sure we can actually stream the number of bytes we are going to stream.

Could you put up this patch to reading the RNTuple anchor as a separate PR? I think this may be something that we even want to backport to the 6.23 branch.

Reply:
https://github.com/root-project/root/pull/15516